### PR TITLE
refactor: fix and clean up ledger polling and status checks

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

...

## Changelog

```
- Please write a list of changes
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

...

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/popups/ManageAccountPopup.svelte [hunk 1]
```diff
@@ -32,14 +28,7 @@
...
 
             isBusy = true
-
-            if ($isActiveLedgerProfile) {
-                promptUserToConnectLedger(_save, _cancel)
-            } else if ($isSoftwareProfile && $isStrongholdLocked) {
-                openPopup({ type: 'password', props: { onSuccess: _save } })
-            } else {
-                void _save()
-            }
```
Comment (MarkNerdi) [inline packages/shared/components/popups/ManageAccountPopup.svelte:42]: don't we need the `checkActiveProfileAuth` here as well?

File: packages/shared/lib/core/profile/actions/active-profile/logout.ts [hunk 2]
```diff
@@ -9,25 +9,27 @@ import { clearPollNetworkInterval } from '@core/network'
...
 
     // (TODO): Figure out why we are using a promise here?
     return new Promise((resolve) => {
         if (_lockStronghold && get(isSoftwareProfile)) {
             lockStronghold()
-        } else if (get(isActiveLedgerProfile)) {
-            stopPollingLedgerStatus()
+        }
+
+        if (isLedgerProfile(type) && isPollingLedgerDeviceStatus) {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/profile/actions/active-profile/logout.ts:31]: there's no way a profile can be a ledger profile and software profile at the same time right? 
If so, `else if`

File: packages/shared/components/atoms/iconButtons/SyncSelectedAccountIconButton.svelte [hunk 3]
```diff
@@ -1,16 +1,10 @@
...
     import { Icon } from 'shared/components'
 
     function handleSyncAccountClick() {
         if (!$selectedAccount.isSyncing) {
-            if ($isSoftwareProfile) {
-                void checkStronghold(syncSelectedAccount)
-            } else {
-                void syncSelectedAccount()
-            }
+            void syncSelectedAccount()
```
Comment (MarkNerdi) [inline packages/shared/components/atoms/iconButtons/SyncSelectedAccountIconButton.svelte:7]: why don't we need a stronghold check here?

**Review by MarkNerdi — CHANGES_REQUESTED** [review]
Works as expected :+1:

File: packages/shared/components/popups/ManageAccountPopup.svelte (hunk 1 above)
Comment (nicole-obrien) [reply packages/shared/components/popups/ManageAccountPopup.svelte:42]: We are not doing anything that requires new address generation or signing.

File: packages/shared/lib/core/profile/actions/active-profile/logout.ts (hunk 2 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/profile/actions/active-profile/logout.ts:31]: I can make this change

File: packages/shared/components/atoms/iconButtons/SyncSelectedAccountIconButton.svelte (hunk 3 above)
Comment (MarkNerdi) [reply packages/shared/components/atoms/iconButtons/SyncSelectedAccountIconButton.svelte:7]: makes sense

**Review by MarkNerdi — APPROVED** [review]

**Pull request merged**